src/main/java/com/onwelo/tinvalidator/validator/LVValidator.java

```diff
@@ -0,0 +1,26 @@
+package com.onwelo.tinvalidator.validator;
+
+import java.util.regex.Pattern;
+
+public class LVValidator implements Validator {
+
+    private static final Pattern PATTERN = Pattern.compile("[0-59]\\d[01]\\d{3}[0-2]\\d{4}");
+
+    public Pattern getPattern(){
+        return PATTERN;
+    }
+
+    public boolean computeControlSum(String tin) {
+        try {
+            if(32 <= Integer.parseInt(tin.substring(0,2))){
```

Comment:
Ten warunek jest dziwny. Przecież nie ma dnia miesiąca większego niż 31, to dalczego zwracamy dla niego informację o poprawnym nipie?
Po drugie liczba dni miesiąca zależy od konkretnego miesiąca. Więc `300211....` nie będzie prawidłową datą a przejdzie naszą walidację.

I sprawdzamy tylko warunek dla osób naturalnych. Całkowicie pominęliśmy warunek dla osób prawnych (to chyba będą firmy) czyli to co nas bardziej interesuje.